expr/expr.go

```diff
@@ -781,44 +784,94 @@ func EvalExpr(e *expr, from, until int32, values map[MetricRequest][]*MetricData
 				}
 				return total[0].Values[i]
 			}
-			var totalString string
 			if e.args[1].etype == etName {
 				totalString = e.args[1].target
 			} else {
 				totalString = fmt.Sprintf("%s(%s)", e.args[1].target, e.args[1].argString)
 			}
-			formatName = func(a *MetricData) string {
-				return fmt.Sprintf("asPercent(%s,%s)", a.Name, totalString)
+			formatName = func(a, b string) string {
+				return fmt.Sprintf("asPercent(%s,%s)", a, b)
+			}
+		} else if len(e.args)%2 == 0 && (e.args[1].etype == etName || e.args[1].etype == etFunc) {
+			total, err := getSeriesArg(e.args[1], from, until, values)
+			if err != nil {
+				return nil, err
+			}
+			if len(total) != len(arg) {
+				return nil, ErrWildcardNotAllowed
+			}
+			multipleSeries = true
+			formatName = func(a, b string) string {
+				return fmt.Sprintf("asPercent(%s,%s)", a, b)
 			}
 		} else {
+			fmt.Printf("%v %v\n", len(e.args), len(arg))
 			return nil, errors.New("total must be either a constant or a series")
 		}
 
 		var results []*MetricData
 
-		for _, a := range arg {
-			r := *a
-			r.Name = formatName(a)
-			r.Values = make([]float64, len(a.Values))
-			r.IsAbsent = make([]bool, len(a.Values))
-			results = append(results, &r)
-		}
+		if multipleSeries {
+			for i := range e.args {
+				if i >= len(e.args)/2 {
```

Comment:
`len(e.args)/2` -- this seems confusing.

Can you document why this is?  It might be nicer to reslice args so you have
```
numerators := e.args[:len(e.args)/2]
denominators := e.args[len(e.args)/2:]
```
and then range over the `numerators` slice.